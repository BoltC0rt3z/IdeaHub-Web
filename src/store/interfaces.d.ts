import { Profile } from './profile/actions/interfaces';

export interface Action {
  type: string;
  payload: any;
}

export interface Store {
  profile: Profile;
}
