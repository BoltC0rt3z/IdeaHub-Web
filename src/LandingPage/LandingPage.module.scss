@import '../styles/styles.module.scss';

.landingPage {
  background: $color-light;
  min-height: 80vh;
  width: 100%;
  padding-bottom: 16px;

  .contentSection {
    padding-top: 32px;
    padding-left: 180px;

    .information {
      .title {
        color: $color-dark-blue-lighter;
        font-weight: normal;
        font-size: 20px;
        margin-bottom: 24px;
      }

      .text {
        color: $color-dark;
        font-size: 16px;
      }
    }

    @include respond-to('md-large') {
      padding-left: 8px;
      padding-right: 8px;
    }
  }

  .form {
    align-content: center;
    padding-left: 8rem;
    padding-right: 32px;

    @include respond-to('xs-m') {
      padding-left: 8px;
    }
  }

  @include respond-to('md-large') {
    padding-left: 16px;
  }
}

.header {
  border-bottom: solid 1px $color-light-gray;
  padding: 16px;
  padding-left: 160px;

  @include respond-to('md-large') {
    padding-left: 8px;
  }
}

.groupSvg {
  width: 100%;
  height: 28rem;

  @include respond-to('xs-l') {
    height: 12rem;
  }
}

.button {
  width: 100%;
}
