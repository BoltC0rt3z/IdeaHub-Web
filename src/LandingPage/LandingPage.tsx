import React, { useState } from "react";
import { Row, Col } from "antd";
import { useMedia } from "react-use";
import { Header, Input, PasswordInput } from "../Components";

import { Button } from "../Components";

import styles from "./LandingPage.module.scss";

const LandingPage: React.FC = () => {
  const isMobile: boolean = useMedia("(max-width: 575px)");
  const [hasAccount, setHasAccount] = useState<boolean>(false);

  const joinNow = (): void => setHasAccount(false);
  const signIn = (): void => setHasAccount(true);

  return (
    <>
      <Row className={styles.header}>
        <Header title="IdeaHub" joinNow={joinNow} signIn={signIn} />
      </Row>
      <Row className={styles.landingPage}>
        <Row type="flex" className={styles.contentSection}>
          <Col span={isMobile ? 24 : 12}>
            <Row className={styles.information}>
              <Col className={styles.title}>POOL RESOURCES</Col>
              <Col className={styles.text}>
                Pool human and monetary resources at your comfort
              </Col>
            </Row>
          </Col>
          <Col span={isMobile ? 24 : 12}>
            <Row className={styles.form}>
              {!hasAccount && (
                <Col style={{ marginBottom: 16 }}>
                  <Input placeholder="Username" />
                </Col>
              )}
              <Col style={{ marginBottom: 16 }}>
                <Input placeholder="Email" />
              </Col>
              {!hasAccount && (
                <Col style={{ marginBottom: 16 }}>
                  <PasswordInput placeholder="Password" />
                </Col>
              )}
              <Col style={{ marginBottom: 40 }}>
                <PasswordInput placeholder="Confirm password" />
              </Col>
              <Col>
                <Button style={{ width: "100%" }} type="primary" size="large">
                  {hasAccount ? "Sign In" : "Join now"}
                </Button>
              </Col>
            </Row>
          </Col>
        </Row>
      </Row>
    </>
  );
};

export default LandingPage;
