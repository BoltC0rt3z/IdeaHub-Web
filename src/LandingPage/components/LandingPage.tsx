import React from "react";
import { Row } from "antd";

import styles from "./LandingPage.module.scss";

interface Props {}

const LandingPage: React.FC<Props> = props => {
  return <Row className={styles.landingPage}>Welcome to IdeaHub</Row>;
};

export default LandingPage;
