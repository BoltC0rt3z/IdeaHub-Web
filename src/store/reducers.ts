import { combineReducers } from 'redux';

import { profile } from './profile';
import { signUp } from './signUp';

export default combineReducers({
  profile,
  signUp,
});
