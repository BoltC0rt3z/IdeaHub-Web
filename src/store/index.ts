import { createStore, applyMiddleware } from 'redux';
import createMiddleWare, { SagaMiddleware } from 'redux-saga';

import reducers from './reducers';

const middleWare: SagaMiddleware = createMiddleWare();

const store = createStore(
  reducers,
  applyMiddleware(middleWare)
);

export default store;
