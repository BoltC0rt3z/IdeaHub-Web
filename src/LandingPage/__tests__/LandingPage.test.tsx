import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import LandingPage from '../LandingPage';
import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../store';

describe('Tests for landing page', () => {
  const wrapper: ReactWrapper = mount(
    <Provider store={store}>
      <Router>
        <LandingPage hasAccount={false} />
      </Router>
    </Provider>,
  );

  it('Should render the landing page without crushing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
