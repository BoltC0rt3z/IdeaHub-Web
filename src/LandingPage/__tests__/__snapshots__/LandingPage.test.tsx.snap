// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for landing page Should render the landing page without crushing 1`] = `ReactWrapper {}`;
